src/engine/enginebuffer.cpp

```diff
@@ -1284,6 +1286,11 @@ bool EngineBuffer::isTrackLoaded() {
     return false;
 }
 
+bool EngineBuffer::isSeekQueued(double* pSeekPosition) {
+    *pSeekPosition = m_queuedSeekPosition.getValue();
```

Comment:
I assume the seek position is only relevant if this function returns true and should not be written otherwise? Then it should be implemented correctly and needs to be documented. This is not obvious.

std::optional would fit here (with a different function name), but it is not available in C++14. You may use a std::pair instead and add a TODO.

Reply:
getQueuedSeekPosition()

Reply:
std::pair is no option for me, because the resulting code is unreadable.
In this case let's backport std::optional

Reply:
`bool getQueuedSeekPosition(double* pPosition)` works for me for the time being

Reply:
I think in this case we need the same memory fence like in process seek.
call m_iSeekQueued.loadAcquire() fist. This should also solve @uklotzde s comment.

Else I have just found a c++11 version of optional:
https://github.com/martinmoene/optional-lite